File: dripy-dropper/src/interfaces/Folder.ts
export interface File {
  fileId: string;
  name: string;
  size: number;
  path: string;
  contentType: string;
  createdAt: string;
}
export interface FolderFiles{
  fileId: string,
  name: string,
  size: 0,
  contentType: string,
  createdAt: string
};


export interface Folder {
  folderId: string;
  name: string;
  files?: FolderFiles[];
  parentFolderId?: string;
  ownerId: string;
}

File: Drippy-Dropper.API/Models/Entities/Folder.cs
namespace Drippy_Dropper.API.Models.Entities
{
    public class Folder
    {
        public Guid FolderId { get; set; }
        public string Name { get; set; }
        public string Path { get; set; }
        public Guid? ParentFolderId { get; set; }
        public Guid OwnerId { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public bool IsActive { get; set; }

        public Folder? ParentFolder { get; set; }
        public ICollection<Folder>? SubFolders { get; set; }
        public ICollection<File>? Files { get; set; }
    }
}
